platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -322,6 +176,75 @@ class MGLStyleValueTransformer {
         getMBGLValue(value.rawValue, mbglValue);
         return mbglValue;
     }
+    
+    NSObject* toRawStyleSpecValue(MGLStyleFunction<ObjCType>* styleFunction) {
+        NSMutableDictionary * rawFunction;
+        // interpolationMode => type
+        switch (styleFunction.interpolationMode) {
+            case MGLInterpolationModeExponential:
+                rawFunction[@"type"] = @"exponential";
+                break;
+            case MGLInterpolationModeInterval:
+                rawFunction[@"type"] = @"interval";
+                break;
+            case MGLInterpolationModeCategorical:
+                rawFunction[@"type"] = @"categorical";
+                break;
+            case MGLInterpolationModeIdentity:
+                rawFunction[@"type"] = @"identity";
+                break;
+        }
+        
+        // defaultValue => default
+        if (styleFunction.defaultValue) {
+            NSCAssert([styleFunction.defaultValue isKindOfClass:[MGLStyleConstantValue class]], @"Default value must be constant");
+            rawFunction[@"default"] = [(MGLStyleConstantValue<ObjCType> *)styleFunction.defaultValue rawValue];
+        }
+        
+        // interpolationBase => base
+        if (styleFunction.interpolationBase) {
+            // TODO: does this direct cast work?
+            rawFunction[@"base"] = (NSNumber *) styleFunction.interpolationBase;
```

Comment:
This cast merely treats a `CGFloat` (`float` or `double`, depending on the architecture) as a pointer to an NSNumber object. To convert a primitive number to an NSNumber object, `@(box)` it or call a method like `+[NSNumber numberWithDouble:]`.